diff -urdN dtao-57fe859b1f24f17a85ee38825159de9eefdfbb10/Makefile dtao-screenpos/Makefile
--- dtao-57fe859b1f24f17a85ee38825159de9eefdfbb10/Makefile	2022-02-16 02:56:16.000000000 +0700
+++ dtao-screenpos/Makefile	2026-09-12 19:21:26.142163532 +0700
@@ -1,20 +1,15 @@
 BINS = dtao
-MANS = doc/dtao.1
 
 PREFIX ?= /usr/local
 CFLAGS += -Wall -Wextra -Wno-unused-parameter -g
 
-all: $(BINS) $(MANS)
+all: $(BINS)
 
 clean:
 	$(RM) $(BINS) $(addsuffix .o,$(BINS))
 
 install: all
-	install -D -t $(PREFIX)/bin $(BINS)
-	install -Dm644 -t $(PREFIX)/share/man/man1 $(MANS)
-
-$(MANS): %: %.ronn
-	ronn -r $<
+	install -D -t $(DESTDIR)/$(PREFIX)/bin $(BINS)
 
 WAYLAND_PROTOCOLS=$(shell pkg-config --variable=pkgdatadir wayland-protocols)
 WAYLAND_SCANNER=$(shell pkg-config --variable=wayland_scanner wayland-scanner)
diff -urdN dtao-57fe859b1f24f17a85ee38825159de9eefdfbb10/doc/dtao.1.ronn dtao-screenpos/doc/dtao.1.ronn
--- dtao-57fe859b1f24f17a85ee38825159de9eefdfbb10/doc/dtao.1.ronn	2022-02-16 02:56:16.000000000 +0700
+++ dtao-screenpos/doc/dtao.1.ronn	2026-09-12 19:21:26.141889741 +0700
@@ -58,9 +58,15 @@
 * `-w` <PIXEL>:
     set window width
 
+* `-x` <PIXEL>:
+    set x position on the screen
+
 * `-xs` <NUM>:
     display on output number <NUM>
 
+* `-y` <PIXEL>:
+    set y position on the screen
+
 * `-z`:
     when used once: bar will not cover "exclusive zones" of other layer-shell surfaces; when used twice: bar will request its own exclusive zone
 
diff -urdN dtao-57fe859b1f24f17a85ee38825159de9eefdfbb10/dtao.c dtao-screenpos/dtao.c
--- dtao-57fe859b1f24f17a85ee38825159de9eefdfbb10/dtao.c	2022-02-16 02:56:16.000000000 +0700
+++ dtao-screenpos/dtao.c	2026-09-12 19:21:26.141952144 +0700
@@ -454,13 +454,11 @@
 	uint32_t anchor = ZWLR_LAYER_SURFACE_V1_ANCHOR_TOP |
 			ZWLR_LAYER_SURFACE_V1_ANCHOR_LEFT |
 			ZWLR_LAYER_SURFACE_V1_ANCHOR_RIGHT;
+	int margintop = 0, marginbottom = 0, marginleft = 0, marginright = 0;
 
 	/* Parse options */
 	for (int i = 1; i < argc; i++) {
-		if (!strcmp(argv[i], "-b")) {
-			anchor ^= ZWLR_LAYER_SURFACE_V1_ANCHOR_TOP |
-				ZWLR_LAYER_SURFACE_V1_ANCHOR_BOTTOM;
-		} else if (!strcmp(argv[i], "-bg")) {
+		if (!strcmp(argv[i], "-bg")) {
 			if (++i >= argc)
 				BARF("option -bg requires an argument");
 			if (parse_color(argv[i], &bgcolor))
@@ -550,6 +548,25 @@
 				BARF("option -xs requires an argument");
 			/* One-based to match dzen2 */
 			output = atoi(argv[i]) - 1;
+		} else if (!strcmp(argv[i], "-x")) {
+			if (++i >= argc)
+				BARF("option -x requires an argument");
+			if (*argv[i] == '-')
+				marginright = atoi(argv[i] + 1);
+			else
+				marginleft = atoi(argv[i]);
+		} else if (!strcmp(argv[i], "-y")) {
+			if (++i >= argc)
+				BARF("option -y requires an argument");
+			if (*argv[i] == '-') {
+				marginbottom = atoi(argv[i] + 1);
+				anchor |= ZWLR_LAYER_SURFACE_V1_ANCHOR_BOTTOM;
+				anchor &= ~ZWLR_LAYER_SURFACE_V1_ANCHOR_TOP;
+			} else {
+				margintop = atoi(argv[i]);
+				anchor |= ZWLR_LAYER_SURFACE_V1_ANCHOR_TOP;
+				anchor &= ~ZWLR_LAYER_SURFACE_V1_ANCHOR_BOTTOM;
+			}
 		} else if (!strcmp(argv[i], "-z")) {
 			exclusive_zone++;
 		} else {
@@ -594,6 +611,8 @@
 
 	zwlr_layer_surface_v1_set_size(layer_surface, width, height);
 	zwlr_layer_surface_v1_set_anchor(layer_surface, anchor);
+	zwlr_layer_surface_v1_set_margin(layer_surface, margintop, marginright,
+			marginbottom, marginleft);
 	wl_surface_commit(wl_surface);
 	wl_display_roundtrip(display);
