--- river/build.zig	2026-03-31 15:14:05.000000000 +0700
+++ river/build.zig.new	2026-04-16 09:43:14.893421666 +0700
@@ -219,7 +219,7 @@
         inline for (.{"river"}) |page| {
             // Workaround for https://github.com/ziglang/zig/issues/16369
             // Even passing a buffer to std.Build.Step.Run appears to be racy and occasionally deadlocks.
-            const scdoc = b.addSystemCommand(&.{ "/bin/sh", "-c", "scdoc < doc/" ++ page ++ ".1.scd" });
+            const scdoc = b.addSystemCommand(&.{ "/bin/sh", "-c", "scdoc < river/doc/" ++ page ++ ".1.scd" });
             // This makes the caching work for the Workaround, and the extra argument is ignored by /bin/sh.
             scdoc.addFileArg(b.path("doc/" ++ page ++ ".1.scd"));
 
